File: CS/LayoutViewCustomButtonContent/MyGridControl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraGrid;

namespace CustomButtonsLayoutView
{
    class MyGridControl : GridControl
    {
        protected override DevExpress.XtraGrid.Views.Base.BaseView CreateDefaultView()
        {
            return CreateView("LayoutViewCustomButton");
        }

        protected override void RegisterAvailableViewsCore(DevExpress.XtraGrid.Registrator.InfoCollection collection)
        {
            base.RegisterAvailableViewsCore(collection);
            collection.Add(new LayoutViewCustomButtonInfoRegistrator());
        }
    }
}

File: VB/LayoutViewCustomButtonContent/MyGridControl.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraGrid

Namespace CustomButtonsLayoutView
    Friend Class MyGridControl
        Inherits GridControl

        Protected Overrides Function CreateDefaultView() As DevExpress.XtraGrid.Views.Base.BaseView
            Return CreateView("LayoutViewCustomButton")
        End Function

        Protected Overrides Sub RegisterAvailableViewsCore(ByVal collection As DevExpress.XtraGrid.Registrator.InfoCollection)
            MyBase.RegisterAvailableViewsCore(collection)
            collection.Add(New LayoutViewCustomButtonInfoRegistrator())
        End Sub
    End Class
End Namespace
